src_features/provideErc20TokenInformation/cmd_provideTokenInfo.c

```diff
@@ -168,45 +168,21 @@ void handleProvideErc20TokenInformation(uint8_t p1,
     }
     if (index < NUM_TOKENS_EXTRA) {
         PRINTF("Descriptor whitelisted\n");
-    } else {
-        cx_ecfp_init_public_key(CX_CURVE_256K1,
-                                LEDGER_SIGNATURE_PUBLIC_KEY,
-                                sizeof(LEDGER_SIGNATURE_PUBLIC_KEY),
-                                &tokenKey);
-        if (!cx_ecdsa_verify(&tokenKey,
-                             CX_LAST,
-                             CX_SHA256,
-                             hash,
-                             32,
-                             workBuffer + offset,
-                             dataLength)) {
+    } else
+#endif
+    {
+        CX_ASSERT(cx_ecfp_init_public_key_no_throw(CX_CURVE_256K1,
+                                                   LEDGER_SIGNATURE_PUBLIC_KEY,
+                                                   sizeof(LEDGER_SIGNATURE_PUBLIC_KEY),
+                                                   &tokenKey));
+        if (!cx_ecdsa_verify_no_throw(&tokenKey, hash, 32, workBuffer + offset, dataLength)) {
```

Comment:
A bit hard to read with all the ifdef
can you please add some comments to help ?

Reply:
In fact, the calls to `cx_ecfp_init_public_key` and `cx_ecdsa_verify` were done twice, with the same parameters, just because of the `#ifdef` which was encapsulating the `if/else.`

The code was:
```
#ifdef HAVE_TOKENS_EXTRA_LIST
...
if (condition) {
    PRINTF()
} else {
    cx_ecfp_init_public_key()
    cx_ecdsa_verify()
}
#else
cx_ecfp_init_public_key()
cx_ecdsa_verify()
#endif
```

Which I simplified with:
```
#ifdef HAVE_TOKENS_EXTRA_LIST
...
if (condition) {
    PRINTF()
} else
#endif
{
    cx_ecfp_init_public_key()
    cx_ecdsa_verify()
}
```

Reply:
Yes sorry we misunderstood: I ended up nderstanding what was done, but I think the code would benefit from comments, preferably a paragraph that explains what is done on each case